### PR TITLE
[Core] React form updates

### PR DESCRIPTION
**Pull request by alisterdev**
- [x] Remove last remaining state from `FileElement`
- [x] Fix bug where `SelectElement` won't clear displayed value
- [x] Remove previously deprecated `HelpTextElement` component
- [x] Fix bug were plain HTML elements were not properly displayed when nested inside `<FormElement/>`
- [x] Fix bug where file name would not display properly in media upload/edit form

### CODE REVIEW

File: jsx/Form.js [hunk 1]
```diff
@@ -214,6 +214,9 @@ var SelectElement = React.createClass({
       elementClass = 'row form-group has-error';
     }
 
+    // After checking for errors, set value to empty string to reset dropdown
+    let value = (this.props.value === undefined) ? "" : this.props.value;
```
Comment (jacobpenny) [inline jsx/Form.js]: Should `const` this up.

File: jsx/Form.js [hunk 2]
```diff
@@ -543,90 +543,68 @@ var FileElement = React.createClass({
...
+    // Add error message
+    if (this.props.hasError) {
+      errorMessage = this.props.errorMessage;
+      elementClass = 'row form-group has-error';
+    }
+
+    // Need to manually reset file value, because HTML API
+    // does not allow setting value to anything than empty string.
+    // Hence can't use value attribute in the input element.
+    var file = document.querySelector(".fileUpload");
```
Comment (jacobpenny) [inline jsx/Form.js]: I realize this code was only moved, but isn't this going to break if there is more than one FileElement on a page?

Comment (alisterdev) [inline jsx/Form.js]: @jacobpenny It's not gonna break, but it's going to work only for the [first upload element].(https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector) I guess it is something we should fix in the future.

**Comment by alisterdev** [comment]
@jacobpenny `const` all changes and fixed a bug found in process

**Pull request merged**